### ISSUE TITLE
The Value of Recall and Lost in the stage of fine-tuning

### ISSUE THREAD

**Issue by zjs2121230251** [cvg/glue-factory#169]
During my reproduction of LightGlue, the recall and loss during the pre-training stage behave normally. However, once I move to the fine-tuning stage, the recall does not improve further, and loss/last also stops decreasing.
I would like to ask whether this behavior is expected or reasonable.

Currently, the metrics are:

**Recall**: 8.881E-01

**loss/last**: 3.210E-01

The detailed outputs are shown below:
`[01/23/2026 10:34:24 gluefactory INFO] [Validation] {match_recall 8.881E-01, match_precision 8.450E-01, accuracy 9.444E-01, average_precision 7.633E-01, loss/total 3.210E-01, loss/last 3.210E-01, loss/assignment_nll 3.210E-01, loss/nll_pos 5.183E-01, loss/nll_neg 1.237E-01, loss/num_matchable 3.656E+02, loss/num_unmatchable 7.110E+02, loss/row_norm 9.146E-01}
[01/23/2026 10:34:26 gluefactory INFO] Finished training on process 1.
[01/23/2026 10:34:28 gluefactory.utils.experiments INFO] Saving checkpoint checkpoint_49_286798.tar
[01/23/2026 10:34:28 gluefactory.utils.experiments INFO] Deleting checkpoint checkpoint_44_258118.tar
[01/23/2026 10:34:28 gluefactory INFO] Finished training on process 0.`

**Comment by benedettilucas** [comment]
Did you perform fine-tuning with frozen weights for the attention/assignment modules, or was it an end-to-end fine-tune using the default repo configs?

**Comment by zjs2121230251** [comment]
> Did you perform fine-tuning with frozen weights for the attention/assignment modules, or was it an end-to-end fine-tune using the default repo configs?

I basically followed the full training pipeline from the repo’s README, so it should fall into the latter case you mentioned.
